agent/recognition/rdd_sampler.py

```
@@ -0,0 +1,164 @@
+        except (TypeError, ValueError):
+            self._interval = 1800.0  # 环境变量非法时兜底，避免 import 阶段崩溃
+        self._last_ts = {}           # key -> 上次落盘时间
+        self._last_hash = {}         # key -> 上次 roi_crop 内容哈希
+
+    # ------------------------------------------------------------------
+    # 对外唯一入口
+    # ------------------------------------------------------------------
+
+    def record(self, *, node, roi, result, stage=None, images=None, meta=None):
+        """
+        采一条样本。node=检测点名；roi=(x,y,w,h) 全局；result="hit"/"miss"；
```

Ruff (0.16.0)

[warning] Docstring contains ambiguous `；` (FULLWIDTH SEMICOLON). Did you mean `;` (SEMICOLON)?

(RUF002)